src/contracts/actions/CappedTokenTransfers.sol

```diff
@@ -8,22 +8,26 @@ import {ICappedTokenTransfers} from 'interfaces/actions/ICappedTokenTransfers.so
 import {IERC20} from 'forge-std/interfaces/IERC20.sol';
 
 contract CappedTokenTransfers is SafeManageable, ICappedTokenTransfers {
-  mapping(address _token => uint256 _transferCap) public tokenCap;
-  mapping(address _token => uint256 _transferCapSpent) public capSpent;
-  mapping(address _token => uint256 _transferCooldown) public tokenCooldown;
-
+  // Token cap configuration
+  mapping(address => uint256) public tokenCap;
+  mapping(address => uint256) public capSpent;
+  mapping(address => uint256) public lastEpochTimestamp;
+  mapping(address => uint256) public tokenEpochLength;
+
+  // State tracking
+  bool public stateUpdated;
   TokenTransfer[] public tokenTransfers;
 
-  constructor(address _safe) SafeManageable(_safe) {}
+  constructor(address _safe) SafeManageable(_safe) {
+    stateUpdated = true;
+  }
```

Comment:
let's add the token and the cap to the constructor 🙏

Reply:
Do you also want to remove the `addCappedToken` function ?

Reply:
yup, immutable is best, 1 token, 1 cap, and if needed we just redeploy and re-whitelist

Reply:
agreed with wei3er